### PR TITLE
[BUG] fix for `update_predict` state handling bug, replace detached cutoff by deepcopy

### LINKED ISSUE #2556: [BUG] `BaseForecaster.update_predict` broken upon second call, `detached_cutoff` does not reset component cutoffs
`update_predict` is broken for all composite forecasters in the following way:

If it is run once for a composite with a component, the `cutoff` of the composite is reset, but not that of the component. This leads to unexpected behaviour and non-compliant predictions when `update_predict` is called a second time.

This is due to a fundamental bug in the design of the `detached_cutoff` that is used internally.

An example is the test failure in https://github.com/alan-turing-institute/sktime/pull/2248, in combination with #2555.

Another minimal example of this behaviour is this, in #2248:

```python
from sktime.utils._testing.series import _make_series
from sktime.forecasting.compose import BaggingForecaster
from sktime.forecasting.model_selection import SlidingWindowSplitter

X = _make_series()
X1 = X[:24]
X2 = X[24:36]
f = BaggingForecaster.create_test_instance()
f.fit(X1, fh=[1, 2])

cv = SlidingWindowSplitter(start_with_window=False)
f.update_predict(X2, cv=cv)
f.update_predict(X2, cv=cv)
```

The expected behaviour is that returns of both `update_predict` are identical, and that the `cutoff` of the internal estimator does not change.

### PR DESCRIPTION
**Pull request by fkiraly**
This PR fixes #2556, the `update_predict` state handling bug, by replacing the `detached_cutoff` by a call of `deepcopy` and working with the deep copy.

This is a temporary fix which ensures correctness - long-term we should probably deprecate the `update_predict` behaviour that resets the `cutoff`.

One secondary problem was that some tests for the online forecaster were *assuming* the behaviour that was problematic in #2556, i.e,. that the inner state *does* get updated by `update_predict`.
This was accounted for by adding a `reset_forecaster` argument to `update_predict`, which can be used to switch between the two behaviours currently assumed in the code base, of `update_predict`.

Docstrings have also been added to clarify what `update_predict` actually does.

### CODE REVIEW

**Comment by fkiraly** [comment]
@GuzalBulatova, I think this will, finally, also fix your bug now!

**Comment by fkiraly** [comment]
this should do it!
https://github.com/alan-turing-institute/sktime/pull/2557

File: sktime/forecasting/base/_base.py [hunk 1]
```diff
@@ -795,6 +798,13 @@ def update_predict(
                 X.index must contain y.index and fh.index both
             there are no restrictions on number of columns (unlike for y)
         update_params : bool, optional (default=True)
+            whether model parameters should be updated in each update step
+        reset_forecaster : bool, optionel (default=True)
```
Comment (ltsaprounis) [inline sktime/forecasting/base/_base.py:802]: typo: optionel -> optional

Comment (ltsaprounis) [inline sktime/forecasting/base/_base.py:802]: typo: optionel -> optional

File: sktime/forecasting/base/_base.py [hunk 2]
```diff
@@ -1984,15 +1977,47 @@ def _predict_proba(self, fh, X, marginal=True):
...
+        cv : temporal cross-validation generator, optional (default=None)
+        X : time series in sktime compatible format, optional (default=None)
+                Exogeneous time series for updating and forecasting
+            Should be of same scitype (Series, Panel, or Hierarchical) as y
+            if self.get_tag("X-y-must-have-same-index"),
+                X.index must contain y.index and fh.index both
+            there are no restrictions on number of columns (unlike for y)
+        update_params : bool, optional (default=True)
+            whether model parameters should be updated in each update step
+        reset_forecaster : bool, optionel (default=True)
```
Comment (ltsaprounis) [inline sktime/forecasting/base/_base.py:2015]: typo

Comment (ltsaprounis) [inline sktime/forecasting/base/_base.py:2015]: typo

**Review by ltsaprounis — APPROVED** [review]
Thanks Franz, that was really confusing me with the composite tests.
Flagged a couple of small typos (optional).

File: sktime/forecasting/base/_base.py (hunk 1 above)
Comment (fkiraly) [reply sktime/forecasting/base/_base.py:802]: fixed

File: sktime/forecasting/base/_base.py (hunk 2 above)
Comment (fkiraly) [reply sktime/forecasting/base/_base.py:2015]: fixed

**Review by ltsaprounis — APPROVED** [review]

**Pull request merged**